xdr/transaction.go

```diff
@@ -0,0 +1,15 @@
+package xdr
+
+// TimeBounds extracts the timebounds (if any) from the transaction's
+// Preconditions.
+func (tx *Transaction) TimeBounds() *TimeBounds {
+	switch tx.Cond.Type {
+	case PreconditionTypePrecondNone:
+		return nil
+	case PreconditionTypePrecondTime:
+		return tx.Cond.TimeBounds
+	case PreconditionTypePrecondV2:
+		return tx.Cond.V2.TimeBounds
+	}
```

Comment:
❗ We should have a panic in here for unrecognized XDR:
```suggestion
		return tx.Cond.V2.TimeBounds
 	default:
		panic("unsupported transaction type: " + e.Type.String())
	}
```